### PR TITLE
storage: fix source reader remap trace compaction, and improve logging

### PR DESCRIPTION
**Pull request by guswynn**
This pr does various things to improve the situation around compaction frontier since's, until we have a fix for the bug figured out. It:

- Makes it so we ACTUALLY compact the `ReclockFollower` trace. Currently we just always compact it to the initial `resume_upper` and never again
- Avoid compacting the follower trace if the `since` hasn't moved
- Share some logic around compaction since calculation
- Add a good number of `info!` log lines around compaction and reclock initialization. As these happen once every 30 seconds and at startup, they should not be too onerous (some happen per-worker).
  - Change the resumption interval to 30 seconds to make logging more reasonable.
  - new output looks like this: https://gist.github.com/guswynn/7a783de31e3a4b5f845a86c72376d491
- Skip the first instantaneous tick in the `resumption_operator` interval to make it more clear what is happening in the pipeline

These combined should make it so we have more clarity around `resumption_frontier`'s and compaction since's, in production, as we watch some of the bandaids and (eventual) fixes we roll out.

### Motivation

  * This PR _is related to_ a recognized bug.

   https://github.com/MaterializeInc/materialize/issues/15402 and incident 6 (cant find an issue)

### Tips for reviewer

This pr has multiple parts to it. If you feel it would be more clear broken into

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/storage/src/source/resumption.rs [hunk 1]
```diff
@@ -77,6 +77,9 @@ where
         // TODO: determine what interval we want here.
         let mut interval = tokio::time::interval(std::time::Duration::from_secs(10));
         interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+        // Clear the first instantaneous tick
+        // See <https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.tick>
+        interval.tick().await;
```
Comment (petrosagg) [inline src/storage/src/source/resumption.rs:82]: Should we call `interval.reset();` instead?

File: src/storage/src/source/source_reader_pipeline.rs [hunk 2]
```diff
@@ -482,7 +489,7 @@ where
                     .await;
 
                 let mut source_upper = reclock_follower
-                    .source_upper_at_frontier(resume_upper.borrow())
+                    .source_upper_at_frontier(initial_resume_upper.borrow())
```
Comment (petrosagg) [inline src/storage/src/source/source_reader_pipeline.rs:492]: nit: Is there a "non-initial" resume upper in this context? Removing the alias from the config destructure seems clearer

File: src/storage/src/source/source_reader_pipeline.rs [hunk 3]
```diff
@@ -546,6 +553,9 @@ where
                 // right now.
                 let mut emit_ts = Timestamp::from((*now_fn)());
 
+                // The last frontier we compacted the remap trace to, starting at [0].
+                let mut last_compaction_since = Antichain::from_elem(Timestamp::default());
```
Comment (petrosagg) [inline src/storage/src/source/source_reader_pipeline.rs:557]: I think `::minimum()` is the correct thing to call here. I haven't seen an instance where it differs from the default but at the same time Frank didn't use the `Default` trait for it for some reason so let's stick with the convention

**Review by petrosagg — COMMENTED** [review]
Do we still need to have the since frontier trail the resume upper by some amount? It seems to me that now that we figured out what happens with the handles and compaction we should be able to compact at whatever the resumption frontier is.

File: src/storage/src/source/source_reader_pipeline.rs (hunk 2 above)
Comment (guswynn) [reply src/storage/src/source/source_reader_pipeline.rs:492]: yeah i can change this back!

**Comment by guswynn** [comment]
@petrosagg we don't need it to trail, but i am going to fix that after we fix the underlying read handle issue!

File: src/storage/src/source/resumption.rs (hunk 1 above)
Comment (guswynn) [reply src/storage/src/source/resumption.rs:82]: i find that less clear from the docs!

**Review by petrosagg — APPROVED** [review]
Sorry I thought I had approved this on Friday. Approving now

**Pull request merged**